app/src/pages/Home.tsx

```diff
@@ -1,39 +1,50 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import Constants from '../constants/Constants';
 import axios from 'axios';
 import RequestsTable from '../components/custom/tables/RequestsTable';
 import { headerFont } from '../constants/fonts';
 import { useAuthService } from '../keycloak';
+import { useNavigate } from 'react-router-dom';
 
 /**
  * @description The Home page, showing a list of reimbursement requests. 
  * @returns A React element
  */
 const Home = () => {
-  const [data, setData] = useState([]);
-  const { BACKEND_URL } = Constants;
+  const [requests, setRequests] = useState([]);
+  const { BACKEND_URL, FRONTEND_URL } = Constants;
   const { state: authState } = useAuthService();
+  const navigate = useNavigate();
 
   // Fires on page load.
   useEffect(() => {
-    (async () => {
-      try {
-        const { data } = await axios.get(`${BACKEND_URL}/api/requests?minimal=true`, {
-          headers: {
-            Authorization : `Bearer ${authState.accessToken}`
-          }
-        })
-        setData(data);
-      } catch (e) {
-        console.warn('Server could not be reached.');
-      }
-    })();
+    const targetPage = sessionStorage.getItem('target-page');
```

Comment:
I love this logic! Its such a simple way of getting around keycloak's redirect policies.

Reply:
Thanks to Brandon for the idea.

Reply:
@dbarkowsky Actually I gave him the idea 😉